$OpenBSD: patch-bgpd_bgp_attr_c,v 1.3 2011/03/26 12:00:53 sthen Exp $

bgpd: attribute jhash call should use a standard interface to in6_addr data
http://code.quagga.net/cgi-bin/gitweb.cgi?p=quagga.git;a=commitdiff;h=98e30f5db6dea65a5d6a176065cd0b60eb7bb001

--- bgpd/bgp_attr.c.orig	Mon Mar 21 14:32:49 2011
+++ bgpd/bgp_attr.c	Sat Mar 26 11:16:31 2011
@@ -373,8 +373,8 @@ attrhash_key_make (void *p)
 
 #ifdef HAVE_IPV6
       MIX(attr->extra->mp_nexthop_len);
-      key = jhash2(attr->extra->mp_nexthop_global.s6_addr32, 4, key);
-      key = jhash2(attr->extra->mp_nexthop_local.s6_addr32, 4, key);
+      key = jhash2(attr->extra->mp_nexthop_global.s6_addr, 16, key);
+      key = jhash2(attr->extra->mp_nexthop_local.s6_addr, 16, key);
 #endif /* HAVE_IPV6 */
     }
 
